pkgs/rbrowser/default.nix

```
@@ -0,0 +1,232 @@
+{ stdenvNoCC
+, writeScriptBin
+, makeDesktopItem
+, withBrave ? true
+, brave
+, withChromium ? true
+, chromium
+, withFirefox ? true
+, firefox
+, rofi
+, zsh
+}:
+
+assert withBrave || withChromium || withFirefox;
+let
+  inherit (stdenvNoCC.lib) optionalString;
+
+
+  rbrowser = writeScriptBin "rbrowser" ''
+    #! ${zsh}/bin/zsh
+    #
+    #  Copyright (c) 2010-2020 Wael Nasreddine <wael.nasreddine@gmail.com>
+    #
+    #  This program is free software; you can redistribute it and/or modify
+    #  it under the terms of the GNU General Public License as published by
+    #  the Free Software Foundation; either version 2 of the License, or
+    #  (at your option) any later version.
+    #
+    #  This program is distributed in the hope that it will be useful,
+    #  but WITHOUT ANY WARRANTY; without even the implied warranty of
+    #  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    #  GNU General Public License for more details.
+    #
+    #  You should have received a copy of the GNU General Public License
+    #  along with this program; if not, write to the Free Software
+    #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+    #  USA.
+    #
+
+    set -euo pipefail
+
+    function help()
+    {
+      echo -e "USAGE: rbrowser [options] <url>"
+      echo -e "\t -h, --help                - this message"
+      echo -e "\t -p, --profile             - the profile to use"
+      echo -e "\t -b, --browser             - the browser to use. Default: chromium"
+    }
+
+    ${optionalString withBrave ''brave="${brave}/bin/brave"''}
+    ${optionalString withChromium ''chromium="${chromium}/bin/chromium"''}
+    ${optionalString withFirefox ''firefox="${firefox}/bin/firefox"''}
+
+    log_depth="0"
+    args=""
+    browser=""
+
+    while [[ $# -ge 1 ]]; do
+      case "''${1}" in
+        -h|--help)
+          help
+          exit 0
+          ;;
+        -p|--profile)
+          if [[ -z "''${2}" ]]; then
```

```suggestion
          if [[ -z "''${2:-}" ]]; then
```
By setting an empty default, bash will do complaining about it being undefined. In talking about the set -u option